packages/devtools_app/lib/src/screens/memory/memory_vm_chart.dart

```diff
@@ -127,6 +127,9 @@ class MemoryVMChartState extends State<MemoryVMChart> with AutoDisposeMixin {
   void didChangeDependencies() {
     super.didChangeDependencies();
 
+    if (_initialized) return;
+    _initialized = true;
```

Comment:
We don't want to return early if we are initialized - we just want to return early if the new controller from provider is the same we already have. Follow the pattern here: https://github.com/flutter/devtools/blob/85a45e5fd7cdf5946da19611dd18513bc28b22bb/packages/devtools_app/lib/src/screens/profiler/profiler_screen.dart#L97-L100

Also, make sure we are calling any code not dependent upon the memory controller before we return early if memoryController has not changed (in this case, set colorScheme before returning early).

Reply:
done